### PR TITLE
[9.0][FIX] Fixed inconsistent UoM domain in Quotations/Order #3

### PR DESCRIPTION
**Pull request by nicknomo**
Description of the issue/feature this PR addresses: 
The product_uom field's domain in sale.order.line would be inconsistent and improper in several situations.

Current behavior before PR:

When adding a line to a quotation/order, the domain of the product_uom would be changed in the onchange function called by a changed product_id. This onchange function would change the domain on product_uom for ALL lines in the order.

Also, when editing the order, the product_uom would have no domain. The user would experience an inconsistent UI compared to when they created the lines.

Desired behavior after PR is merged:

The category for the product_uom's domain will always be properly defined. Each line will have an independent domain, and it will restrict the product_uom selection so that it can only be changed to UoM's in the same category as the existing UoM.

To solve these issues:

1) I Added a relative field for the category_id of product_uom, and stored it in the database.

2) I added this relative field to the view (invisible=1), and used it in the default domain for product_uom.

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by nicknomo** [comment]
CLA try #2

File: addons/sale/sale.py [hunk 1]
```diff
@@ -661,6 +661,8 @@ def write(self, values):
     product_id = fields.Many2one('product.product', string='Product', domain=[('sale_ok', '=', True)], change_default=True, ondelete='restrict', required=True)
     product_uom_qty = fields.Float(string='Quantity', digits=dp.get_precision('Product Unit of Measure'), required=True, default=1.0)
     product_uom = fields.Many2one('product.uom', string='Unit of Measure', required=True)
+    #We've added a related field to store the category of the UoM for this line. We use this as a default domain in the view.
+    relcatid = fields.Many2one(related='product_uom.category_id', store=True)
```
Comment (mart-e) [inline addons/sale/sale.py]: not compatible with [stable policy](https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean)

**Comment by mart-e** [comment]
Hello,

We can not add new fields in stable versions. Please either resubmit for the master branch or adapt your code to avoid the new field.

**Comment by C3POdoo** [comment]
Dear @nicknomo,

Thank you for your contribution but the version 9.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**